### PR TITLE
feat: implement flutter web apis with new Amplitude-JavaScript interfaces and implement identify/identifyGroup

### PR DESCRIPTION
**Pull request by yuhao900914**
This pr includes:
1. update ios example to make it works
2. update the Amplitude-JavaScript script and implement flutter web apis with new Amplitude-JavaScript interfaces + implement identify/identifyGroup, 
3. add missed startNewSession parameter in setUserId on andorid.

### CODE REVIEW

File: example/lib/user_id_form.dart [hunk 1]
```diff
@@ -16,9 +16,24 @@ class _UserIdFormState extends State<UserIdForm> {
 
   @override
   Widget build(BuildContext context) {
-    return TextField(
-        autocorrect: false,
-        decoration: InputDecoration(labelText: 'User Id'),
-        onChanged: makeHandler(context));
+    return Column(
```
Comment (yuhao900914) [inline example/lib/user_id_form.dart:19]: Add the test case for getUserId method.

File: example/web/index.html [hunk 2]
```diff
@@ -30,47 +30,113 @@
...
-          }
-          ; var l = ["setProductId", "setQuantity", "setPrice", "setRevenueType", "setEventProperties"]
-          ; for (var p = 0; p < l.length; p++) { s(u, l[p]) } n.Revenue = u
-          ; var d = ["init", "logEvent", "logRevenue", "setUserId", "setUserProperties", "setOptOut", "setVersionName", "setDomain", "setDeviceId", "enableTracking", "setGlobalUserProperties", "identify", "clearUserProperties", "setGroup", "logRevenueV2", "regenerateDeviceId", "groupIdentify", "onInit", "logEventWithTimestamp", "logEventWithGroups", "setSessionId", "resetSessionId"]
-          ; function v(e) {
-            function t(t) {
-        e[t] = function () {
-          e._q.push([t].concat(Array.prototype.slice.call(arguments, 0)))
-
+    (function (window, document) {
```
Comment (yuhao900914) [inline example/web/index.html]: update the Amplitude-JavaScript script with the newest version.

File: example/ios/Podfile [hunk 3]
```diff
@@ -10,80 +10,32 @@ project 'Runner', {
   'Release' => :release,
 }
 
-def parse_KV_file(file, separator='=')
-  file_abs_path = File.expand_path(file)
-  if !File.exists? file_abs_path
-    return [];
+def flutter_root
```
Comment (yuhao900914) [inline example/ios/Podfile:13]: iOS flutter has build failed problem when test. 
Those files are auto changed by removing the Podfile, Podfile.lock and running `flutter build ios`. 
https://stackoverflow.com/questions/65516334/flutter-error-invalid-podfile-file-no-implicit-conversion-of-nil-into-string

File: example/lib/my_app.dart [hunk 4]
```diff
@@ -96,8 +97,6 @@ class _MyAppState extends State<MyApp> {
                 divider,
                 GroupIdentifyForm(),
                 divider,
-                GroupIdentifyForm(),
```
Comment (yuhao900914) [inline example/lib/my_app.dart:99]: duplicate test.

File: example/ios/Flutter/AppFrameworkInfo.plist [hunk 5]
```diff
@@ -21,6 +21,6 @@
   <key>CFBundleVersion</key>
   <string>1.0</string>
   <key>MinimumOSVersion</key>
-  <string>8.0</string>
+  <string>9.0</string>
```
Comment (qingzhuozhen) [inline example/ios/Flutter/AppFrameworkInfo.plist:24]: Any affect on this version change?

File: example/web/index.html (hunk 2 above)
Comment (qingzhuozhen) [reply example/web/index.html]: Can we use minified version?

File: example/ios/Flutter/AppFrameworkInfo.plist (hunk 5 above)
Comment (yuhao900914) [reply example/ios/Flutter/AppFrameworkInfo.plist:24]: This change related to I updated the flutter version to the latest one which is v2.5.3. It's auto-updated the MiniumnOSVersion from 8.0 to 9.0 when you run `flutter build ios`. 
I don't think this version update will affect anything.
1. this is the version update in the example app. 
2. I also tried to convert the version back to stable v2.0.0(the version starts to support flutter [web),](https://medium.com/flutter/whats-new-in-flutter-2-0-fe8e95ecc65) everything works fine and this change won't be included.

**Review by qingzhuozhen — APPROVED** [review]

**Pull request merged**